docs/rfc/0003-collaborative-objects.md

```diff
@@ -0,0 +1,727 @@
+pointing to that object at `refs/namespaces/<project>/rad/collaborative-objects/<typename>/<object ID>`, 
+where `object ID` is a unique identifier generated at creation time. 
+
+This allows us to fetch subsets of collaborative objects by specifying refspecs
+that match them. The downside is that we are adding a ref to the initial
+advertised refs, each of theses refs is around 250 bytes. If we consider a popular
+repository such as https://github.com/facebook/react/ you can see that they have
+nearly 10,000 issues (including both Open and Closed, which we must). That would 
+mean that the initial ref advertisment in any replication of this repository
+would be ~2.5Mb. If we could use v2 of the git pack protocol this problem would
+go away but alas, we must use v1. 
+
+To get around the problem we can add a parameter to the radicle URL which
+indicates either a single object ID or a type name which we wish to fetch
+from the remote, this would then be passed as a custom header to the transport
+and interpreted on the server. In this manner we can allow clients to choose
+when they want to replicate collaborative objects, which would allow staged
+fetches where we first fetch the repository and identities, and then fetch 
+collaborative objects.
+
+Therefore we add two parameters to a radicle URL:
+
+- `collab_object_id`: specifies that a server should only consider references to the given object ID
+- `collab_type_name`: specifies that a server should only consider references to the given type name
+
+These parameters are emitted by the radicle p2p transport as headers in a 
+similar fashion to the `n=` parameter for the nonce or the `ls=true`
+parameter for selecting the `upload-pack-ls` service. The Radicle git server
+can then use these parameters to filter the refs it operates on.
+
+
+### Updating objects
+
+To make a change to an object we load the existing messages for an object. The 
+application developer provides us with the binary representation of the change
+to that object. We apply the change and ensure that the new object state still
+matches the object schema. At this point the state of the object may depend on
+many contributions from the tracking graph - not just the ones in our own view
+of the project. We now create a commit with our new change in it, referencing
+all the commits containing the direct dependencies of the change as parents.
+
+### Schema Commits
+
+Schemas are important for the interoperability of the system. We need
+applications to be able to rely on the data they are working with being valid,
+otherwise we impose the problem of schema validation on application developers.
+
+Schemas will need to be able to change over time. Schema migration is out of 
+scope for this RFC but we need a minimal mechanism to support it in future. To
+this end schemas are represented using their own hash graph. For the purposes
+of this RFC a schema is a commit with a tree that contains a single `schema.json`
```

Comment:
Where are these commits going to be found?

Reply:
It's mentioned earlier in the document that they are referenced by a commit trailer on changes, I'll add a sentence to that effect here as well.